claims.go

```diff
@@ -9,7 +9,7 @@ import (
 // Claims must just have a Valid method that determines
 // if the token is invalid for any supported reason
 type Claims interface {
-	Valid() error
+	Valid(options ...*ClaimsValidationOptions) error
```

Comment:
As @mfridman already mentioned, would it be possible to turn these into functional style options? Similar to what we have in https://github.com/golang-jwt/jwt/blob/main/parser_option.go

Reply:
The `WithLeeway` function is functional style https://github.com/golang-jwt/jwt/pull/139/files#diff-72f644d282bce7e136d0f762de1b9d5f3e447bfa88e5e0c4974ac2479107ddc9R33-R37 

It is far easier to set this as an attribute of the parser which is what the old library did, but it is not a backward compatible change and given how the `Valid` function is called (https://github.com/golang-jwt/jwt/pull/139/files#diff-83eb8e32639d01cf443d6d8bde24c1c8be78766090d8c5f8586c36250cfedca6R87)  I am frankly unsure how you make the claims validation functional style, however, I would be happy to make the change with your input.